storage/src/main/java/zipkin2/storage/kafka/KafkaAutocompleteTags.java

```diff
@@ -34,62 +42,105 @@
  * for scatter gather data from different instances.
  */
 public class KafkaAutocompleteTags implements AutocompleteTags {
+  static final String HTTP_BASE_URL = "http://%s:%d";
+
   final KafkaStreams traceStoreStream;
 
   KafkaAutocompleteTags(KafkaStorage storage) {
     traceStoreStream = storage.getTraceStoreStream();
   }
 
   @Override public Call<List<String>> getKeys() {
-    ReadOnlyKeyValueStore<String, Set<String>> autocompleteTagsStore =
-        traceStoreStream.store(AUTOCOMPLETE_TAGS_STORE_NAME,
-            QueryableStoreTypes.keyValueStore());
-    return new GetKeysCall(autocompleteTagsStore);
+    return new GetTagKeysCall(traceStoreStream);
   }
 
   @Override public Call<List<String>> getValues(String key) {
-    ReadOnlyKeyValueStore<String, Set<String>> autocompleteTagsStore =
-        traceStoreStream.store(AUTOCOMPLETE_TAGS_STORE_NAME,
-            QueryableStoreTypes.keyValueStore());
-    return new GetValuesCall(autocompleteTagsStore, key);
+    return new GetTagValuesCall(traceStoreStream, key);
   }
 
-  static class GetKeysCall extends KafkaStreamsStoreCall<List<String>> {
-    final ReadOnlyKeyValueStore<String, Set<String>> autocompleteTagsStore;
+  static class GetTagKeysCall extends Call.Base<List<String>> {
+    static final Gson GSON = new Gson();
+
+    final KafkaStreams traceStoreStream;
 
-    GetKeysCall(ReadOnlyKeyValueStore<String, Set<String>> autocompleteTagsStore) {
-      this.autocompleteTagsStore = autocompleteTagsStore;
+    GetTagKeysCall(KafkaStreams traceStoreStream) {
+      this.traceStoreStream = traceStoreStream;
     }
 
-    @Override protected List<String> query() {
-      List<String> keys = new ArrayList<>();
-      autocompleteTagsStore.all().forEachRemaining(keyValue -> keys.add(keyValue.key));
-      return keys;
+    @Override protected List<String> doExecute() throws IOException {
+      return traceStoreStream.allMetadataForStore(AUTOCOMPLETE_TAGS_STORE_NAME)
+          .parallelStream()
```

Comment:
Instead of `parallelStream` which tasks these onto the common forkjoin pool, we can just use armeria's asynchronousness. I think parallelStream partitions into num cpus * 2, I'm not sure if you explicitly want that, if you do it might be better to stick with it than this suggestion which will execute all requests in parallel.

```java
CompletableFuture<List<AggregatedHttpResponse>> responses = CompletableFuture.allOf(
  traceStoreStream.map(...).map(client.get.aggregate()).collect(toList));
return responses.join().stream().checkResponses/deserialize
```

Reply:
didn't thought much about this. giving a try :)